adr/2025_03_10-suggestions/README.md

```diff
@@ -0,0 +1,172 @@
+# Versioning & Suggestions
+
+Due to the similarity between document versioning, red lining, track changes and suggestions, it is worth seeing an overview of all of their requirements to make sure we consider solutions that resolve multiple of them.
+
+- A diff between two versions of a document, is essentially a set of suggestions that takes your previous document, and transforms it into the more recent document.
+- Red-lining and suggestions are the same, red-lining just allows you to attach metadata to the suggestion, like a comment of why the change was made.
+- Track changes, is entering a mode where any edit you make, becomes a suggestion to the document.
+
+## Versioning Overview
+
+Versioning has already been implemented for Y.js & y-prosemirror, but to be explicit about the sort of thing we are looking for it is good to explicitly state the feature set versioning should have:
+
+- A user can take _snapshots_ of the current document state, and store that in a separate data store
+- These snapshots can be viewed independently at any point in time later
+- These snapshots can be _compared_ with each other
+  - This allows the user to see what changed between the two snapshots
+  - Changes should show:
+    - The content that was changed (i.e. additions or deletions)
+    - Updates to content (i.e. changing an attribute)
+    - Which user identifiers modified the content
+    - Ideally, a timestamp of when the content was modified
+    - Changes should only show content that was present in both documents (i.e. if some content was added and subsequently deleted in some version between the two snapshots, because it is not present in the final document of either snapshot, it should not show)
+  - These changes can be visualized in a number of ways: as a unified diff, split diff, or even stacked diff. _See [Appendix](#displaying-changes) for more information_
+
+**Prior Art:**
+
+- Google Docs [Video here]
```

Comment:
maybe also link to ink and switch projects to indicate how big the problem / solution space can be:
- https://www.inkandswitch.com/upwelling/
- https://www.inkandswitch.com/patchwork/notebook/
-